src/main/java/org/jitsi/jigasi/transcription/TranscriptHandler.java

```diff
@@ -17,130 +17,103 @@
  */
 package org.jitsi.jigasi.transcription;
 
+import net.java.sip.communicator.service.protocol.*;
+
 import java.util.*;
 
 /**
- * A TranscriptHandler can build a Transcript in a desired format and
- * publish it to a desired location
- *
- * @param <T> the type wherein the transcript will be made
+ * This class is used to publish TranscriptionResults and Transcript to the
+ * what is desired by adding the necessary {@link TranscriptionResultPublisher}
+ * and {@link TranscriptPublisher}s
  *
  * @author Nik Vaessen
  */
-public interface TranscriptHandler<T>
+public class TranscriptHandler
 {
     /**
-     * Start building a formatted transcript
-     *
-     * @return the builder
+     * The list of {@link TranscriptionResultPublisher} which will handle
+     * {@link TranscriptionResult}s
      */
-    Formatter<T> format();
+    private List<TranscriptionResultPublisher> resultPublishers
+        = new LinkedList<>();
 
     /**
-     * Format a single result into
-     *
-     * @param result the result
-     * @return the formatted result
+     * The list of {@link TranscriptPublisher} which will handle
+     * {@link Transcript}s
      */
-    T formatTranscriptionResult(TranscriptionResult result);
+    private List<TranscriptPublisher> transcriptPublishers = new LinkedList<>();
 
     /**
-     * Publish a formatted transcript to a desired location
+     * Handle a {@link TranscriptionResult} with all given
+     * {@link TranscriptionResultPublisher}'s
      *
-     * @param transcript the formatted transcript to publish
+     * @param room the {@link ChatRoom} to send the result to
+     * @param result the {@link TranscriptionResult} to handle
      */
-    void publish(T transcript);
+    public void publishTranscriptionResult(ChatRoom room,
+                                           TranscriptionResult result)
+    {
+        for(TranscriptionResultPublisher p : resultPublishers)
+        {
+            p.publish(room, result);
+        }
+    }
 
     /**
-     * The Formatter which used a fluent API
+     * Get the {@link TranscriptPublisher.Promise}s which can handle a
+     * {@link Transcript} for all given {@link TranscriptPublisher}s
      *
-     * @param <T> the type wherein the transcript will be formatted
+     * @return a list of {@link TranscriptPublisher.Promise}s
      */
-    interface Formatter<T>
+    public List<TranscriptPublisher.Promise> getTranscriptPublishPromises()
```

Comment:
I don't really see a great way how to do it otherwise... How would you combine multiple descriptions?